joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/cpp/features17/Cpp17FeaturesTests.scala

```diff
@@ -30,6 +54,38 @@ class Cpp17FeaturesTests extends AstC2CpgSuite(fileSuffix = FileDefaults.CppExt)
       c2.typeFullName shouldBe "MyContainer"
       // We are unable to express this template argument deduction in the current schema
       cpg.typeDecl.member.nameExact("val").typeFullName.l shouldBe List("T")
+
+      val List(c1Method, c2Method) = cpg.typeDecl.nameExact("MyContainer").astChildren.isMethod.isConstructor.l
+      c1Method.fullName shouldBe "MyContainer.MyContainer:void()"
+      c1Method.signature shouldBe "void()"
+      c2Method.fullName shouldBe "MyContainer.MyContainer:void(T)"
+      c2Method.signature shouldBe "void(T)"
+      c2Method.body.astChildren.isCall.isAssignment.code.l shouldBe List("this->val = val")
+
+      cpg.call.isAssignment.code.l shouldBe List(
+        "this->val = val",
+        "c1 = new MyContainer.MyContainer(1)",
+        "c2 = new MyContainer.MyContainer()"
+      )
+      cpg.call.isAssignment.argument(1).isIdentifier.typeFullName.l shouldBe List("MyContainer", "MyContainer")
+
+      val List(c1NewCall, c2NewCall) = cpg.call.isAssignment.argument(2).isCall.l
+
+      c1NewCall.code shouldBe "new MyContainer.MyContainer(1)"
+      c1NewCall.methodFullName shouldBe Defines.OperatorNew
```

Comment:
Where is this coming from? There is no `new` in the code. The `new` operator results in a heap allocation. It's very important that we don't mix those up with stack/inline allocations.

(Also in other tests, which I'm not all flagging individually.)